packages/java/endpoint/src/main/java/com/vaadin/hilla/signals/NumberSignal.java

```diff
@@ -0,0 +1,163 @@
+package com.vaadin.hilla.signals;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.vaadin.hilla.signals.core.StateEvent;
+import jakarta.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * A signal that holds a number value.
+ */
+public class NumberSignal {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(NumberSignal.class);
+
+    private final ReentrantLock lock = new ReentrantLock();
+
+    private final UUID id = UUID.randomUUID();
+
+    private Double value;
+
+    private final Set<Sinks.Many<ObjectNode>> subscribers = new HashSet<>();
+
+    /**
+     * Creates a new NumberSignal with the provided default value.
+     *
+     * @param defaultValue
+     *            the default value
+     */
+    public NumberSignal(@Nullable Double defaultValue) {
+        this.value = defaultValue;
+    }
+
+    /**
+     * Creates a new NumberSignal with the default value of 0.
+     */
+    public NumberSignal() {
+        this(0.0);
+    }
+
+    /**
+     * Subscribes to the signal.
+     *
+     * @return a Flux of JSON events
+     */
+    public Flux<ObjectNode> subscribe() {
+        Sinks.Many<ObjectNode> sink = Sinks.many().multicast()
```

Comment:
Good point, you're right. 
Basically, mulitcast is for having one Flux with multiple subscribers, but we cannot use here, since the behaviour of the Flux when an error happens is to propagate the error to all subscribers, and then to terminate the Flux (no more propagations can happen through that). 
We need a separate Sink (and Flux) for each client, so that when an error happens in propagating the updates to one client, it doesn't terminate the flow of updates to other clients. So, each sink has only one subscriber in practice, and unicast the correct way to do it.

Reply:
Done.